jupyter_collaboration/rooms.py

```diff
@@ -100,6 +100,7 @@ async def initialize(self) -> None:
             # try to apply Y updates from the YStore for this document
             read_from_source = True
             if self.ystore is not None:
+                await self.ystore.started.wait()
```

Comment:
@jzhang20133 You didn't apply my [suggestion](https://github.com/jupyterlab/jupyter-collaboration/pull/299#discussion_r1588855135).

Reply:
@davidbrochart Following your suggestion, we moved the logic here in document room. By not applying your suggestion, are you referring to this line`self.create_task(ystore.start())`?  I did not add ystore.start logic here because ystore.start() is also called in yroom.broadcast_update() method. And if ystore has been started and then it is started again, [an exception could be thrown](https://github.com/jupyter-server/pycrdt-websocket/blob/main/pycrdt_websocket/ystore.py#L354).   Instead of triggering starting of ystore at multiple places and exception handling at multiple places, we choose to simply wait for the ystore to be started, which is a great idea that you have suggested.

Reply:
Thanks, @jzhang20133. I'll let David respond to this when he's back in office next week and we can create a follow-up PR if needed.

@davidbrochart's point below stands. If we do something different from his suggestions, we should include an explanation and wait for his response before merging. 

This one is on me folks. I missed this difference in the change before approving the PR. Let's learn from this and make sure we're properly patient going forward.

Reply:
@Zsailer @davidbrochart I will make sure I add explanation if change is made differently from suggested and wait for @davidbrochart 's inputs and approval on the difference in the future PRs.  Thank you both for your patience and help on reviewing PRs! Really appreciated!!!

Reply:
@davidbrochart  For this specific change, please feel free to let us know your thought here about not adding this line `self.create_task(ystore.start())`. I can address accordingly in another PR.

Reply:
yes, it just wait for it to be started by yroom.broadcast_update() method. If yroom is started twice, [an exception could be thrown](https://github.com/jupyter-server/pycrdt-websocket/blob/main/pycrdt_websocket/ystore.py#L354) and the exception could kill task groups. We need to add exception handling for ystore.start() if it get started in multiple places.